plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/edit.tsx

```diff
@@ -55,12 +57,25 @@ const INNER_BLOCKS_TEMPLATE: InnerBlockTemplate[] = [
 	],
 ];
 
-// Get product type options.
-const productTypes = getProductTypeOptions();
-
 const AddToCartOptionsEdit = ( props: BlockEditProps< Attributes > ) => {
 	const { setAttributes } = props;
 
+	const { productTypes, currentProduct } = useSelect< {
+		productTypes: ProductTypeProps[];
+		currentProduct: ProductTypeProps;
+	} >( ( select ) => {
+		const { getProductTypes, getCurrentProductType } = select(
+			woocommerceTemplateStateStore
+		);
+
+		return {
+			productTypes: getProductTypes(),
+			currentProduct: getCurrentProductType(),
```

Comment:
No need to do it in this PR, but in the future we will need to make sure we have a default product type. I expect that when we don't have product context, it will be `Simple Product`, and when we have product context, we will use the one from the product (and ideally hide this control?).

Reply:
Currently, if the product type is non, or even one, the toolbar button isn't rendered. Do you think we should render something

Reply:
I mean that before any is selected, we will be defaulting, in most cases, to `Simple Product`, no? In that case, shouldn't the button have the `Simple Product` text so it's clear the Simple Product state is being edited? This is how it works in the Cart block, instead of having a `Switch state` button, it shows by default `Filled cart`. :thinking:

Reply:
yes, gotcha. Let's handle in https://github.com/woocommerce/woocommerce/pull/53430